python/cudf/cudf/core/column/decimal.py

```diff
@@ -364,18 +364,45 @@ def _get_decimal_type(lhs_dtype, rhs_dtype, op):
     else:
         raise NotImplementedError()
 
+    if isinstance(lhs_dtype, type(rhs_dtype)):
+        # SCENARIO 1: If `lhs_dtype` & `rhs_dtype` are same, then try to
+        # see if `precision` & `scale` can be fit into this type.
+        try:
+            return lhs_dtype.__class__(precision=precision, scale=scale)
+        except ValueError:
+            # Call to _validate fails, which means we need
+            # to goto SCENARIO 3.
+            pass
+    else:
+        # SCENARIO 2: If `lhs_dtype` & `rhs_dtype` are of different dtypes,
+        # then try to see if `precision` & `scale` can be fit into the type
+        # with greater MAX_PRECISION (i.e., the bigger dtype).
+        try:
+            if lhs_dtype.MAX_PRECISION >= rhs_dtype.MAX_PRECISION:
+                return lhs_dtype.__class__(precision=precision, scale=scale)
+            else:
+                return rhs_dtype.__class__(precision=precision, scale=scale)
+        except ValueError:
+            # Call to _validate fails, which means we need
+            # to goto SCENARIO 3.
+            pass
```

Comment:
Put the `try` on the outside, since the exception handling is the same for all cases. This reduces 5 lines of code.
```suggestion
    try:
    if isinstance(lhs_dtype, type(rhs_dtype)):
        # SCENARIO 1: If `lhs_dtype` & `rhs_dtype` are same, then try to
        # see if `precision` & `scale` can be fit into this type.
        return lhs_dtype.__class__(precision=precision, scale=scale)
    else:
        # SCENARIO 2: If `lhs_dtype` & `rhs_dtype` are of different dtypes,
        # then try to see if `precision` & `scale` can be fit into the type
        # with greater MAX_PRECISION (i.e., the bigger dtype).
        if lhs_dtype.MAX_PRECISION >= rhs_dtype.MAX_PRECISION:
            return lhs_dtype.__class__(precision=precision, scale=scale)
        else:
            return rhs_dtype.__class__(precision=precision, scale=scale)
    except ValueError:
        # Call to _validate fails, which means we need
        # to goto SCENARIO 3.
        pass
```